React/Modules/RCTUIManager.m

```diff
@@ -723,6 +727,20 @@ - (void)_amendPendingUIBlocksWithStylePropagationUpdateForShadowView:(RCTShadowV
   }
 }
 
+- (void)addUIManagerObserver:(id<RCTUIManagerObserver>)observer
+{
+  [_uiManagerObserversLock lock];
```

Comment:
Yea I see, so now it's a race instead of happening all the time. Anyway if I make the change to use one observer this locking won't be needed anymore.